### PR TITLE
feat(components): add `MainModal`, `OpenMainModal` and `CloseMainModal`

### PR DESCRIPTION
**Pull request by javieri-empathy**
EX-6546

- Adds the `XModal` component, extracted from the archetype.
- Adds `OpenXModal` and `CloseXModal` components so no events configuration is needed.
- Add wires to reset each module state.

## Known issues

- Closing and opening does not work right now. The `MainScroll` component fails to create the `IntersectionObserver`. After further investigation the `hasContent` helper is causing this issue. Removing this helper from the used layout makes it work.

### CODE REVIEW

File: packages/x-components/src/components/modals/__tests__/x-modal.spec.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import { XEventsTypes } from '../../../wiring/events.types';
+
+/**
+ * Mounts a {@link XModal} component with the provided options and offers an API to easily
+ * test it.
+ *
+ * @param options - The options to render the component with.
+ * @returns An API to test the component.
+ */
+function mountXModal({ template = '<XModal/>' }: MountXModalOptions = {}): MountXModalAPI {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/x-modal.spec.ts:15]: `mount` or `render`? I don't mind changing to `mount`, but I think we have set it to `render` in our own naming convention 😅 

Same with the `options` and `API` names

```suggestion
function renderXModal({ template = '<XModal/>' }: RenderXModalOptions = {}): RenderXModalAPI {
```

File: packages/x-components/src/components/modals/open-x-modal.vue [hunk 2]
```diff
@@ -0,0 +1,67 @@
+<template>
+  <BaseEventsModalOpen v-on="$listeners" class="x-modal" :opening-event="openingEvent">
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/open-x-modal.vue:2]: I think we are using `camelCase` for the `props`. I think webstorm's autocomplete uses the `kebab-case` approach. Can you refactor them?

```suggestion
  <BaseEventsModalOpen v-on="$listeners" class="x-modal" :openingEvent="openingEvent">
```

File: packages/x-components/src/components/modals/close-x-modal.vue [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<template>
+  <BaseEventsModalClose v-on="$listeners" class="x-modal" :closing-event="closingEvent">
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/close-x-modal.vue:2]: I think we are using `camelCase` for the `props`. I think webstorm's autocomplete uses the `kebab-case` approach. Can you refactor them?

```suggestion
  <BaseEventsModalClose v-on="$listeners" class="x-modal" :closingEvent="closingEvent">
```

File: packages/x-components/src/x-modules/history-queries/wiring.ts [hunk 4]
```diff
@@ -151,5 +151,8 @@ export const historyQueriesWiring = createWiring({
   },
   UserClickedConfirmDisableHistoryQueries: {
     setHistoryQueriesDisabled
+  },
+  UserClickedCloseX: {
+    clearHistoryQueriesQuery
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/history-queries/wiring.ts:156]: Shouldn't we add the same wires to the `UserClickedOutOfXModal` `XEvent`?

```suggestion
  UserClickedCloseX: {
    clearHistoryQueriesQuery
  },
  UserClickedOutOfXModal: {
    clearHistoryQueriesQuery
```

File: packages/x-components/src/x-modules/identifier-results/wiring.ts [hunk 5]
```diff
@@ -121,5 +121,8 @@ export const identifierResultsWiring = createWiring({
   },
   ExtraParamsChanged: {
     setIdentifierResultsExtraParams
+  },
+  UserClickedCloseX: {
+    clearIdentifierResultsQuery
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/identifier-results/wiring.ts:126]:
```suggestion
  },
  UserClickedCloseX: {
    clearIdentifierResultsQuery
  },
  UserClickedOutOfXModal: {
    clearIdentifierResultsQuery
```

File: packages/x-components/src/x-modules/query-suggestions/wiring.ts [hunk 6]
```diff
@@ -112,5 +112,8 @@ export const querySuggestionsWiring = createWiring({
   },
   ExtraParamsChanged: {
     setQuerySuggestionsExtraParams
+  },
+  UserClickedCloseX: {
+    clearQuerySuggestionsQuery
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/query-suggestions/wiring.ts:117]:
```suggestion
  },
  UserClickedCloseX: {
    clearQuerySuggestionsQuery
  },
  UserClickedOutOfXModal: {
    clearQuerySuggestionsQuery
```

File: packages/x-components/src/x-modules/related-tags/wiring.ts [hunk 7]
```diff
@@ -118,5 +118,8 @@ export const relatedTagsWiring = createWiring({
   },
   ExtraParamsChanged: {
     setRelatedTagsExtraParams
+  },
+  UserClickedCloseX: {
+    clearRelatedTagsQuery
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/related-tags/wiring.ts:123]:
```suggestion
  },
  UserClickedCloseX: {
    clearRelatedTagsQuery
  },
  UserClickedOutOfXModal: {
    clearRelatedTagsQuery
```

File: packages/x-components/src/x-modules/search-box/wiring.ts [hunk 8]
```diff
@@ -63,5 +63,8 @@ export const searchBoxWiring = createWiring({
   },
   UserPressedClearSearchBoxButton: {
     clearSearchBoxQuery
+  },
+  UserClickedCloseX: {
+    clearSearchBoxQuery
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/search-box/wiring.ts:68]:
```suggestion
  },
  UserClickedCloseX: {
    clearSearchBoxQuery
  },
  UserClickedOutOfXModal: {
    clearSearchBoxQuery
```

File: packages/x-components/src/x-modules/search/wiring.ts [hunk 9]
```diff
@@ -192,5 +199,8 @@ export const searchWiring = createWiring({
   },
   ExtraParamsChanged: {
     setSearchExtraParams
+  },
+  UserClickedCloseX: {
+    clearSearchQuery
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/search/wiring.ts:204]:
```suggestion
  },
  UserClickedCloseX: {
    clearSearchQuery
  },
  UserClickedOutOfXModal: {
    clearSearchQuery
```

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]
good job!

File: packages/x-components/src/x-modules/history-queries/wiring.ts (hunk 4 above)
Comment (javieri-empathy) [reply packages/x-components/src/x-modules/history-queries/wiring.ts:156]: Luis, fuck, you are right. Sorry and thanks for realising about it.

**Review by LuisMartinez15 — APPROVED** [review]
good job!

File: packages/x-components/src/components/modals/x-modal.vue [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+   * A specialised version of a modal component, made to contain a full search application.
+   *
+   * @public
+   */
+  @Component({
+    components: {
+      BaseEventsModal
+    }
+  })
+  export default class XModal extends Vue {
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-components/src/components/modals/x-modal.vue:30]: And about the Events names, we also should rename them, but let's wait until we have the modals Xmodule, because we have to rethink also the repeated Events for the same (Open/Close main modal). At the moment I would only rename the XModal

File: packages/x-components/src/views/home/Home.vue [hunk 11]
```diff
@@ -5,7 +5,7 @@
     <PreselectedFilters />
     <UrlHandler query="q" store="store" />
     <SnippetCallbacks />
-    <BaseEventButton data-test="open-modal" :events="eventsToOpenX">Start</BaseEventButton>
+    <OpenXModal data-test="open-modal">Start</OpenXModal>
```
Comment (tajespasarela) [inline packages/x-components/src/views/home/Home.vue:8]: I would move the `data-test` to the `OpenXModal` component

File: packages/x-components/src/components/modals/x-modal.vue (hunk 10 above)
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-components/src/components/modals/x-modal.vue:30]: Also I would rename the `OpenXModal` and `CloseXModal` components

File: packages/x-components/src/x-modules/history-queries/wiring.ts (hunk 4 above)
Comment (tajespasarela) [reply packages/x-components/src/x-modules/history-queries/wiring.ts:156]: Yes, when we move the modals to its own XModule, we have to rethink all these events

File: packages/x-components/src/components/modals/__tests__/x-modal.spec.ts [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+  const parent = document.createElement('div');
+  document.body.appendChild(parent);
+
+  const wrapper = mount(
+    {
+      template,
+      components: { XModal }
+    },
+    {
+      attachTo: parent,
```
Comment (tajespasarela) [inline packages/x-components/src/components/modals/__tests__/x-modal.spec.ts:26]: Why is this necessary? I removed it and the tests are passing in any case.
Add a little comment explaining why.

Also, according to the doc, you can use the body directly, and a div will be created automatically:
image

**Review by tajespasarela — CHANGES_REQUESTED** [review]

**Review by tajespasarela — APPROVED** [review]

File: packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+import { installNewXPlugin } from '../../../__tests__/utils';
+import CloseMainModal from '../close-main-modal.vue';
+
+/**
+ * Renders the {@link CloseMainModal} with the provided options.
+ *
+ * @param options - The options to render the component with.
+ * @returns An small API to test the component.
+ */
+function renderCloseXModal({
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts:13]:
```suggestion
function renderCloseMainModal({
```

File: packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+    const { click } = renderCloseXModal({
+      template: '<CloseMainModal @click="onClick">Close</CloseMainModal>',
+      methods
+    });
+    click();
+    expect(methods.onClick).toHaveBeenCalledTimes(1);
+  });
+});
+
+interface RenderCloseXModalOptions {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts:71]:
```suggestion
interface RenderCloseMainModalOptions {
```

File: packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+  /** The template to render. */
+  template?: string;
+  /** Additional methods to add to the testing template. */
+  methods?: Record<string, AnyFunction>;
+}
+
+interface RenderCloseXModalAPI {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts:78]:
```suggestion
interface RenderCloseMainModalAPI {
```

File: packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+  return {
+    wrapper,
+    async click() {
+      await wrapper.trigger('click');
+    }
+  };
+}
+
+describe('testing Close X Modal button component', () => {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts:39]:
```suggestion
describe('testing Close Main Modal button component', () => {
```

File: packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+/**
+ * Renders the {@link CloseMainModal} with the provided options.
+ *
+ * @param options - The options to render the component with.
+ * @returns An small API to test the component.
+ */
+function renderCloseXModal({
+  template = '<CloseMainModal />',
+  methods
+}: RenderCloseXModalOptions = {}): RenderCloseXModalAPI {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts:16]:
```suggestion
}: RenderCloseMainModalOptions = {}): RenderCloseMainModalAPI {
```

File: packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+    const onUserClickedCloseX = jest.fn();
+    wrapper.vm.$x.on('UserClickedCloseX').subscribe(onUserClickedCloseX);
+
+    await click();
+
+    expect(onUserClickedCloseX).toHaveBeenCalledTimes(1);
+  });
+
+  it('renders the default slot contents', () => {
+    const { wrapper } = renderCloseXModal({
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts:51]:
```suggestion
    const { wrapper } = renderCloseMainModal({
```

File: packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+    wrapper,
+    async click() {
+      await wrapper.trigger('click');
+    }
+  };
+}
+
+describe('testing Close X Modal button component', () => {
+  it('emits UserClickedCloseX by default when clicked', async () => {
+    const { wrapper, click } = renderCloseXModal();
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts:41]:
```suggestion
    const { wrapper, click } = renderCloseMainModal();
```

File: packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+    });
+
+    expect(wrapper.text()).toEqual('Close');
+  });
+
+  it('can be extended adding listeners', () => {
+    const methods = {
+      onClick: jest.fn()
+    };
+    const { click } = renderCloseXModal({
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/close-main-modal.spec.ts:62]:
```suggestion
    const { click } = renderCloseMainModal({
```

File: packages/x-components/src/components/modals/__tests__/main-modal.spec.ts [hunk 21]
```diff
@@ -0,0 +1,115 @@
...
+import { XEventsTypes } from '../../../wiring/events.types';
+
+/**
+ * Renders a {@link MainModal} component with the provided options and offers an API to easily
+ * test it.
+ *
+ * @param options - The options to render the component with.
+ * @returns An API to test the component.
+ */
+function renderXModal({ template = '<MainModal/>' }: RenderXModalOptions = {}): RenderXModalAPI {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/main-modal.spec.ts:15]:
```suggestion
function renderMainModal({ template = '<MainModal/>' }: RenderMainModalOptions = {}): RenderMainModalAPI {
```

File: packages/x-components/src/components/modals/__tests__/main-modal.spec.ts [hunk 22]
```diff
@@ -0,0 +1,115 @@
...
+      return wrapper.find(getDataTestSelector('modal-content'));
+    },
+    async focusOutOfModal() {
+      document.body.dispatchEvent(new FocusEvent('focusin'));
+      await localVue.nextTick();
+    }
+  };
+}
+
+describe('testing X Modal  component', () => {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/main-modal.spec.ts:50]:
```suggestion
describe('testing Main Modal  component', () => {
```

File: packages/x-components/src/components/modals/__tests__/main-modal.spec.ts [hunk 23]
```diff
@@ -0,0 +1,115 @@
...
+  it('opens and closes when UserClickedOpenX and UserClickedClosedX are emitted', async () => {
+    const { emit, getModalContent } = renderXModal();
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/main-modal.spec.ts:52]:
```suggestion
    const { emit, getModalContent } = renderMainModal();
```

File: packages/x-components/src/components/modals/__tests__/main-modal.spec.ts [hunk 24]
```diff
@@ -0,0 +1,115 @@
...
+
+    await emit('UserClickedOpenX');
+    expect(getModalContent().exists()).toBe(true);
+
+    await emit('UserClickedCloseX');
+    expect(getModalContent().exists()).toBe(false);
+  });
+
+  it('closes when clicking on the modal overlay', async () => {
+    const { clickModalOverlay, emit, getModalContent } = renderXModal();
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/main-modal.spec.ts:63]:
```suggestion
    const { clickModalOverlay, emit, getModalContent } = renderMainModal();
```

File: packages/x-components/src/components/modals/__tests__/main-modal.spec.ts [hunk 25]
```diff
@@ -0,0 +1,115 @@
...
+
+    await emit('UserClickedOpenX');
+    expect(getModalContent().exists()).toBe(true);
+
+    await clickModalOverlay();
+    expect(getModalContent().exists()).toBe(false);
+  });
+
+  it('closes when focusing any other element out of the modal', async () => {
+    const { emit, focusOutOfModal, getModalContent } = renderXModal();
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/main-modal.spec.ts:73]:
```suggestion
    const { emit, focusOutOfModal, getModalContent } = renderMainModal();
```

File: packages/x-components/src/components/modals/__tests__/main-modal.spec.ts [hunk 26]
```diff
@@ -0,0 +1,115 @@
...
+
+    await emit('UserClickedOpenX');
+    expect(getModalContent().exists()).toBe(true);
+
+    await focusOutOfModal();
+    expect(getModalContent().exists()).toBe(false);
+  });
+
+  it('does not close when clicking inside the content', async () => {
+    const { wrapper, emit, getModalContent } = renderXModal({
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/main-modal.spec.ts:83]:
```suggestion
    const { wrapper, emit, getModalContent } = renderMainModal({
```

File: packages/x-components/src/components/modals/__tests__/main-modal.spec.ts [hunk 27]
```diff
@@ -0,0 +1,115 @@
...
+
+    await emit('UserClickedOpenX');
+    expect(getModalContent().exists()).toBe(true);
+
+    await wrapper.find(getDataTestSelector('test-button')).trigger('click');
+    expect(getModalContent().exists()).toBe(true);
+  });
+});
+
+interface RenderXModalOptions {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/main-modal.spec.ts:99]:
```suggestion
interface RenderMainModalOptions {
```

File: packages/x-components/src/components/modals/__tests__/main-modal.spec.ts [hunk 28]
```diff
@@ -0,0 +1,115 @@
...
+  /** The template to render. */
+  template?: string;
+}
+
+interface RenderXModalAPI {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/main-modal.spec.ts:104]:
```suggestion
interface RenderMainModalAPI {
```

File: packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts [hunk 29]
```diff
@@ -0,0 +1,83 @@
...
+import { installNewXPlugin } from '../../../__tests__/utils';
+import OpenMainModal from '../open-main-modal.vue';
+
+/**
+ * Renders the {@link OpenMainModal} with the provided options.
+ *
+ * @param options - The options to render the component with.
+ * @returns An small API to test the component.
+ */
+function renderOpenXModal({
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts:13]:
```suggestion
function renderOpenMainModal({
```

File: packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts [hunk 30]
```diff
@@ -0,0 +1,83 @@
...
+  template = '<OpenMainModal />',
+  methods
+}: RenderOpenXModalOptions = {}): RenderOpenXModalAPI {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts:16]:
```suggestion
}: RenderOpenMainModalOptions = {}): RenderOpenMainModalAPI {
```

File: packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts [hunk 31]
```diff
@@ -0,0 +1,83 @@
...
+
+  return {
+    wrapper,
+    async click() {
+      await wrapper.trigger('click');
+    }
+  };
+}
+
+describe('testing Open X Modal button component', () => {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts:39]:
```suggestion
describe('testing Open Main Modal button component', () => {
```

File: packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts [hunk 32]
```diff
@@ -0,0 +1,83 @@
...
+  it('emits UserClickedOpenX by default when clicked', async () => {
+    const { wrapper, click } = renderOpenXModal();
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts:41]:
```suggestion
    const { wrapper, click } = renderOpenMainModal();
```

File: packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts [hunk 33]
```diff
@@ -0,0 +1,83 @@
...
+    const onUserClickedOpenX = jest.fn();
+    wrapper.vm.$x.on('UserClickedOpenX').subscribe(onUserClickedOpenX);
+
+    await click();
+
+    expect(onUserClickedOpenX).toHaveBeenCalledTimes(1);
+  });
+
+  it('renders the default slot contents', () => {
+    const { wrapper } = renderOpenXModal({
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts:51]:
```suggestion
    const { wrapper } = renderOpenMainModal({
```

File: packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts [hunk 34]
```diff
@@ -0,0 +1,83 @@
...
+    });
+
+    expect(wrapper.text()).toEqual('Open');
+  });
+
+  it('can be extended adding listeners', () => {
+    const methods = {
+      onClick: jest.fn()
+    };
+    const { click } = renderOpenXModal({
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts:62]:
```suggestion
    const { click } = renderOpenMainModal({
```

File: packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts [hunk 35]
```diff
@@ -0,0 +1,83 @@
...
+      template: '<OpenMainModal @click="onClick">Open</OpenMainModal>',
+      methods
+    });
+    click();
+    expect(methods.onClick).toHaveBeenCalledTimes(1);
+  });
+});
+
+interface RenderOpenXModalOptions {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts:71]:
```suggestion
interface RenderOpenMainModalOptions {
```

File: packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts [hunk 36]
```diff
@@ -0,0 +1,83 @@
...
+  /** The template to render. */
+  template?: string;
+  /** Additional methods to add to the testing template. */
+  methods?: Record<string, AnyFunction>;
+}
+
+interface RenderOpenXModalAPI {
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/__tests__/open-main-modal.spec.ts:78]:
```suggestion
interface RenderOpenMainModalAPI {
```

File: packages/x-components/src/components/modals/close-main-modal.vue [hunk 37]
```diff
@@ -0,0 +1,72 @@
...
+<docs lang="mdx">
+## Events
+
+This component emits the following events:
+
+- [`UserClickedCloseX`](./../../api/x-components.xeventstypes.md)
+
+## See it in action
+
+Here you have a basic example of how the close x modal button works.
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/close-main-modal.vue:47]:
```suggestion
Here you have a basic example of how the close main modal button works.
```

File: packages/x-components/src/components/modals/main-modal.vue [hunk 38]
```diff
@@ -0,0 +1,95 @@
...
+  export default class MainModal extends Vue {
+    /**
+     * Animation to use for opening/closing the modal.
+     *
+     * @public
+     */
+    @Prop()
+    public animation?: Vue | string;
+    /**
+     * Events to listen for opening the x modal.
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/main-modal.vue:40]:
```suggestion
     * Events to listen for opening the modal.
```

File: packages/x-components/src/components/modals/main-modal.vue [hunk 39]
```diff
@@ -0,0 +1,95 @@
...
+     *
+     * @internal
+     */
+    protected openEvents: XEvent[] = ['UserClickedOpenX', 'UserOpenXProgrammatically'];
+    /**
+     * Events to listen for closing the x modal.
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/main-modal.vue:46]:
```suggestion
     * Events to listen for closing the modal.
```

File: packages/x-components/src/components/modals/main-modal.vue [hunk 40]
```diff
@@ -0,0 +1,95 @@
...
+<docs lang="mdx">
+## Events
+
+This component emits the following events:
+
+- [`UserClickedOutOfMainModal`](./../../api/x-components.xeventstypes.md)
+
+## See it in action
+
+Here you have a basic example of how the x modal works.
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/main-modal.vue:70]:
```suggestion
Here you have a basic example of how the main modal works.
```

File: packages/x-components/src/components/modals/open-main-modal.vue [hunk 41]
```diff
@@ -0,0 +1,72 @@
...
+<docs lang="mdx">
+## Events
+
+This component emits the following events:
+
+- [`UserClickedOpenX`](./../../api/x-components.xeventstypes.md)
+
+## See it in action
+
+Here you have a basic example of how the open x modal button works.
```
Comment (LuisMartinez15) [inline packages/x-components/src/components/modals/open-main-modal.vue:47]:
```suggestion
Here you have a basic example of how the open main modal button works.
```

File: packages/x-components/src/wiring/events.types.ts [hunk 42]
```diff
@@ -107,6 +107,11 @@ export interface XEventsTypes
    * Payload: none.
    */
   UserClickedCloseX: void;
+  /**
+   * The user clicked out of the Main modal
```
Comment (LuisMartinez15) [inline packages/x-components/src/wiring/events.types.ts:111]:
```suggestion
   * The user clicked out of the main modal.
```

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]

**Review by tajespasarela — APPROVED** [review]

**Review by LuisMartinez15 — APPROVED** [review]
